pydis_site/apps/api/models/bot/role.py

```diff
@@ -43,7 +45,15 @@ class Role(ModelReprMixin, models.Model):
         ),
         help_text="The integer value of the permission bitset of this role from Discord."
     )
+    position = models.IntegerField(
+        default=-1,
```

Comment:
Why does this field need to have a default value?

Reply:
To account for existing rows in the database that do not yet have a value for `position`. Another option would be to allow the field to be null, but that would mean the value returned by the ORM could be `None` and necessitates more complex logic for the `__lt__` method to account for that. (We can't compare `None` to an `int` with the `<` operator).

I figured a better alternative would be to use a default value that should be lower than the lowest role in the hierarchy (which has a position of `0`; although there's no formal lower bound in the Discord API documentation and the discord.py docs mention that the position is "usually positive") and was compatible with the desired methods. 

Hence, an integer value of `-1`.

Reply:
Which entries wouldn't have a position? Old ones that are imported? Apart from that I don't see why this can't be a required field.

Reply:
This migration will be applied to databases that already have entries in them that don't have a position field, so don't have a value for position. Django won't migrate if we don't provide it with a value or allow Null values. The latter would mean more complicated logic, as then it would theoretically be possible for the ORM to give back `NoneType` values instead of `int`.

Reply:
Setting a default would make this field always optional, right? If that's the case then, if possible, it'd be better to manually give a default value during migration to only what's currently in the database so this field will still be required for new records.

Reply:
I've solved it in a different way. Basically, I first add the field with a default value, to ensure that all the existing rows have a non-null value in that column, and then alter the field to remove the default value. 

This is basically the same methodology as the intermediate, temporary field in the link @MarkKoz gave me, but without the actual temporary field to back-up existing data (which we don't have, since the field does not exist before this migration)/creating a PrimaryKey relation using that backed-up data.